omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainSubResourceController.java

```diff
@@ -1,168 +1,204 @@
+		Object created = res.create(parentUuid, post, context);
+		return RestUtil.created(response, created);
+	}
+	
+	/**
+	 * @param parentUuid
+	 * @param uuid
+	 * @param post
+	 * @param request
+	 * @param response
+	 * @return
+	 * @throws Exception
+	 */
+	@RequestMapping(value = "/{resource}/{parentUuid}/{subResource}/{uuid}", method = RequestMethod.POST)
+	@ResponseBody
+	public Object update(@PathVariable("resource") String resource, @PathVariable("parentUuid") String parentUuid,
+	        @PathVariable("subResource") String subResource, @PathVariable("uuid") String uuid,
+	        @RequestBody SimpleObject post, HttpServletRequest request, HttpServletResponse response)
+	        throws ResponseException {
+		baseUriSetup.setup(request);
+		RequestContext context = RestUtil.getRequestContext(request, response);
+		SubResource res = (SubResource) restService.getResourceByName(buildResourceName(resource) + "/" + subResource);
+		Object updated = res.update(parentUuid, uuid, post, context);
+		return RestUtil.updated(response, updated);
+	}
+	
+	/**
+	 * @param parentUuid
+	 * @param uuid
+	 * @param reason
+	 * @param request
+	 * @throws Exception
+	 */
+	@RequestMapping(value = "/{resource}/{parentUuid}/{subResource}/{uuid}", method = RequestMethod.DELETE, params = "!purge")
+	@ResponseBody
+	public Object delete(@PathVariable("resource") String resource, @PathVariable("parentUuid") String parentUuid,
+	        @PathVariable("subResource") String subResource, @PathVariable("uuid") String uuid,
+	        @RequestParam(value = "reason", defaultValue = "web service call") String reason, HttpServletRequest request,
+	        HttpServletResponse response) throws ResponseException {
+		baseUriSetup.setup(request);
+		RequestContext context = RestUtil.getRequestContext(request, response);
+		SubResource res = (SubResource) restService.getResourceByName(buildResourceName(resource) + "/" + subResource);
+		res.delete(parentUuid, uuid, reason, context);
+		return RestUtil.noContent(response);
+	}
+	
+	/**
+	 * @param parentUuid
+	 * @param uuid
+	 * @param request
+	 * @param response
+	 * @throws Exception
+	 */
+	@RequestMapping(value = "/{resource}/{parentUuid}/{subResource}/{uuid}", method = RequestMethod.DELETE, params = "purge")
+	@ResponseBody
+	public Object purge(@PathVariable("resource") String resource, @PathVariable("parentUuid") String parentUuid,
+	        @PathVariable("subResource") String subResource, @PathVariable("uuid") String uuid, HttpServletRequest request,
+	        HttpServletResponse response) throws ResponseException {
+		baseUriSetup.setup(request);
+		RequestContext context = RestUtil.getRequestContext(request, response);
+		SubResource res = (SubResource) restService.getResourceByName(buildResourceName(resource) + "/" + subResource);
+		res.purge(parentUuid, uuid, context);
+		return RestUtil.noContent(response);
+	}
+	
+	/**
```

Comment:
changes I have made are from this point onwards. The original file ends at line 168.